src/environment/stella_environment.cpp

```diff
@@ -195,6 +203,26 @@ bool StellaEnvironment::isTerminal() const {
      m_state.getEpisodeFrameNumber() >= m_max_num_frames_per_episode));
 }
 
+void StellaEnvironment::pressSelect(size_t num_steps) {
+  Event* event = m_osystem->event();
+  m_state.pressSelect(event);
+  for (size_t t = 0; t < num_steps; t++) {
+    m_osystem->console().mediaSource().update();
+  }
+  processScreen();
```

Comment:
Are the process() calls needed?

Reply:
After the `process()` is called in `pressSelect` we call `emulate`, which call the `process()` again. So, maybe not in `pressSelect`? However, we don't call `emulate` in `setDifficulty`. Honestly, I don't know. Any suggestions on how I can figure this out for sure?